### PR TITLE
Spacevine Tweaks

### LINKED ISSUE #19058: Scythes don't work on kudzu
Neither do welders. I'm talking unmutated, plain old green kudzu. Why is a hatchet and a flamer the only counter for a potentially round ending weed? Also bluespace vines growing into windows aren't killable unless you break the window.

### PR DESCRIPTION
**Pull request by silicons**
Kudzu now has a health system.
Scythes now work as a viable defense on these special snowflake plants
Welders now work as they were meant to
Kudzu eradication is no longer based on RNG.
:cl:
tweak: Kudzu is no longer unstoppable
/:cl:

Will fix #19058 .

### CODE REVIEW

**Comment by Zilenan91** [comment]
it's about time someone did something about it :+1:

File: code/modules/events/spacevine.dm [hunk 1]
```diff
@@ -308,15 +308,12 @@
...
-/datum/spacevine_mutation/woodening/on_hit(obj/effect/spacevine/holder, mob/hitter, obj/item/I, expected_damage)
-	if(!expected_damage)
-		return
-	if(hitter)
-		if(I)
-			. = I.force * 2
-		else
-			. = 8
+/datum/spacevine_mutation/woodening/on_hit(obj/effect/spacevine/holder, mob/living/hitter, obj/item/I, expected_damage)
+	if(I.is_sharp())
```
Comment (ChangelingRain) [inline code/modules/events/spacevine.dm]: If it's not sharp but is hot, it probably won't take damage. It should take damage from hot objects, at least if it doesn't have fireproof.

Comment (kevinz000) [inline code/modules/events/spacevine.dm]: checking when i get to my computer :^)

**Comment by kevinz000** [comment]
hi im done it works properly now

File: code/modules/events/spacevine.dm [hunk 2]
```diff
@@ -44,7 +44,7 @@
 	return
 
 /datum/spacevine_mutation/proc/on_hit(obj/effect/spacevine/holder, mob/hitter, obj/item/I, expected_damage)
-	. = expected_damage
+	return expected_damage
```
Comment (ChangelingRain) [inline code/modules/events/spacevine.dm]: You don't need to change this

Comment (kevinz000) [inline code/modules/events/spacevine.dm]: Reese ok il fix it in a bit

**Comment by optimumtact** [comment]
mistakes were made

File: code/modules/events/spacevine.dm (hunk 1 above)
Comment (optimumtact) [inline code/modules/events/spacevine.dm]: did you check?

**Pull request merged**